health/health_json.c

```diff
@@ -184,6 +184,9 @@ static inline void health_rrdcalc2json_nolock(RRDHOST *host, BUFFER *wb, RRDCALC
         if(rc->dimensions && *rc->dimensions)
             health_string2json(wb, "\t\t\t", "lookup_dimensions", rc->dimensions, ",\n");
 
+        if(rc->foreachdim && *rc->foreachdim)
+            health_string2json(wb, "\t\t\t", "foreach", rc->foreachdim, ",\n");
```

Comment:
I don't understand why, but I don't get this string in the response:
```
{
	"hostname": "arch-esxi",
	"latest_alarm_log_unique_id": 1560020145,
	"status": true,
	"now": 1569409900,
	"alarms": {
		"example.random.example_random3": {
			"id": 1569402711,
			"name": "example_random3",
			"chart": "example.random",
			"family": "random",
			"active": true,
			"disabled": false,
			"silenced": false,
			"exec": "/usr/libexec/netdata/plugins.d/alarm-notify.sh",
			"recipient": "sysadmin",
			"source": "2@/etc/netdata/health.d/example.conf",
			"units": "random number",
			"info": "",
			"status": "WARNING",
			"last_status_change": 1569409900,
			"last_updated": 1569409900,
			"next_update": 1569409902,
			"update_every": 2,
			"delay_up_duration": 0,
			"delay_down_duration": 0,
			"delay_max_duration": 0,
			"delay_multiplier": 1.000000,
			"delay": 0,
			"delay_up_to_timestamp": 1569409900,
			"warn_repeat_every": "0",
			"crit_repeat_every": "0",
			"value_string": "91 random number",
			"last_repeat": "0",
			"lookup_dimensions":"random3",
			"db_after": 1569409899,
			"db_before": 1569409899,
			"lookup_method": "average",
			"lookup_after": -1,
			"lookup_before": 0,
			"lookup_options": "",
			"warn":"$this &gt; (($status &gt;= $WARNING) ? (50) : (80))",
			"warn_parsed":"(${this} &gt; ((${status} &gt;= ${WARNING}) ? 50 : 80))",
			"crit":"$this &gt; (($status == $CRITICAL) ? (80) : (95))",
			"crit_parsed":"(${this} &gt; ((${status} == ${CRITICAL}) ? 80 : 95))",
			"green":null,
			"red":null,
			"value":91
		}
	}
}
```

Reply:
A very good find, thank you very much.
This is a dead code, when I started to develop I was keeping the `foreachdim` when I created the alarm, but after to exchange some messages with Ilya(Thank you Ilya), I was convinced that the `foreach` is a kind of template, and this specific variable is not more filled for normal alarms, they continue there only with their "templates".
I removed this code with the last commit.